seatunnel-connectors-v2/connector-flink-table-store/src/main/java/org/apache/seatunnel/connectors/seatunnel/fts/source/FlinkTableStoreSourceSplitEnumerator.java

```diff
@@ -0,0 +1,160 @@
+    private final Table table;
+
+    public FlinkTableStoreSourceSplitEnumerator(
+            Context<FlinkTableStoreSourceSplit> context, Table table) {
+        this.context = context;
+        this.table = table;
+        this.assignedSplit = new HashSet<>();
+    }
+
+    public FlinkTableStoreSourceSplitEnumerator(
+            Context<FlinkTableStoreSourceSplit> context,
+            Table table,
+            FlinkTableStoreSourceState sourceState) {
+        this.context = context;
+        this.table = table;
+        this.assignedSplit = sourceState.getAssignedSplits();
+    }
+
+    @Override
+    public void open() {
+        this.pendingSplit = new HashSet<>();
+    }
+
+    @Override
+    public void run() throws Exception {
+        // do nothing
+    }
+
+    @Override
+    public void close() throws IOException {
+        // do nothing
+    }
+
+    @Override
+    public void addSplitsBack(List<FlinkTableStoreSourceSplit> splits, int subtaskId) {
+        if (!splits.isEmpty()) {
+            pendingSplit.addAll(splits);
+            assignSplit(subtaskId);
+        }
+    }
+
+    @Override
+    public int currentUnassignedSplitSize() {
+        return pendingSplit.size();
+    }
+
+    @Override
+    public void registerReader(int subtaskId) {
+        pendingSplit = getTableSplits();
+        assignSplit(subtaskId);
+    }
+
+    @Override
+    public FlinkTableStoreSourceState snapshotState(long checkpointId) throws Exception {
+        return new FlinkTableStoreSourceState(assignedSplit);
+    }
+
+    @Override
+    public void notifyCheckpointComplete(long checkpointId) throws Exception {
+        // do nothing
+    }
+
+    @Override
+    public void handleSplitRequest(int subtaskId) {
+        // do nothing
+    }
+
+    /** Assign split by reader task id */
+    private void assignSplit(int taskId) {
+        ArrayList<FlinkTableStoreSourceSplit> currentTaskSplits = new ArrayList<>();
+        if (context.currentParallelism() == 1) {
+            // if parallelism == 1, we should assign all the splits to reader
+            currentTaskSplits.addAll(pendingSplit);
+        } else {
+            // if parallelism > 1, according to hashCode of split's id to determine whether to
+            // allocate the current task
+            for (FlinkTableStoreSourceSplit fileSourceSplit : pendingSplit) {
+                final int splitOwner =
+                        getSplitOwner(fileSourceSplit.splitId(), context.currentParallelism());
+                if (splitOwner == taskId) {
+                    currentTaskSplits.add(fileSourceSplit);
+                }
+            }
+        }
+        // assign splits
+        context.assignSplit(taskId, currentTaskSplits);
+        // save the state of assigned splits
+        assignedSplit.addAll(currentTaskSplits);
+        // remove the assigned splits from pending splits
+        currentTaskSplits.forEach(split -> pendingSplit.remove(split));
```

Comment:
Does `currentTaskSplits` need to store in state?

Reply:
Not need, SourceSplitEnumerator has no interface to store state.